citadel/indico_citadel/backend.py

```diff
@@ -0,0 +1,178 @@
+                response = session.delete(url, json=json)
+
+        if not response.ok:
+            raise Exception(f'{response.status_code} - {response.text} in record {json}')
+        elif change_type & SimpleChange.created:
+            content = response.json()
+            CitadelSearchAppIdMap.create(content['metadata']['control_number'], entry.id, entry_type)
+        response.close()
+
+    def upload_file(self, entry, entries, session):
+        with entry.attachment.file.open() as file:
+            response = session.put(
+                url_join(self.search_app, f'api/record/{entry.search_id}/files/attachment'),
+                data=file
+            )
+        if response.ok:
+            entry.attachment_file_id = entry.attachment.file.id
+            db.session.merge(entry)
+            db.session.commit()
+        else:
+            self.logger.error('Failed uploading attachment %d: %s', entry.attachment.id, response.text)
+
+    def run_initial(self, events, total=None, progress=True):
+        cte = Category.get_tree_cte(lambda cat: db.func.json_build_object('id', cat.id, 'title', cat.title))
+        self.categories = dict(db.session.execute(select([cte.c.id, cte.c.path])).fetchall())
+        super().run_initial(events, total)
+
+    def upload_records(self, records, from_queue):
+        session = requests.Session()
+        retry = Retry(
+            total=5,
+            backoff_factor=30,
+            status_forcelist=[500, 502, 503, 504],
+            allowed_methods=frozenset(['POST', 'PUT', 'DELETE'])
+        )
+        session.mount(self.search_app, HTTPAdapter(max_retries=retry))
+        session.headers = self.headers
+        uploader = parallelize(self.upload_record, entries=records)
+        uploader(session, from_queue)
+
+    def upload_files(self, files):
+        session = requests.Session()
+        retry = Retry(
+            total=5,
+            backoff_factor=30,
+            status_forcelist=[500, 502, 503, 504],
+            allowed_methods=frozenset(['PUT'])
+        )
+        session.mount(self.search_app, HTTPAdapter(max_retries=retry))
+        session.headers = self.headers
+        uploader = parallelize(self.upload_file, entries=files, batch_size=100)
+        uploader(session)
+
+
+class LiveSyncCitadelBackend(LiveSyncBackendBase):
+    """Citadel
+
+    This backend uploads data to Citadel.
+    """
+
+    uploader = LiveSyncCitadelUploader
+    unique = True
+
+    def run_export_files(self, batch=1000, force=False):
+        from indico_citadel.plugin import CitadelPlugin
+
+        attachments = (
+            CitadelSearchAppIdMap.query
+            .join(Attachment)
+            .join(AttachmentFile, Attachment.file_id == AttachmentFile.id)
+            .filter(Attachment.type == AttachmentType.file)
+            .filter(AttachmentFile.size <= 10 * 1024 * 1024)
+            .filter(AttachmentFile.extension.in_([s.lower() for s in CitadelPlugin.settings.get('file_extensions')]))
```

Comment:
```suggestion
            .filter(db.func.lower(AttachmentFile.extension).in_([s.lower() for s in CitadelPlugin.settings.get('file_extensions')]))
```

(I don't think the stored extensions are guaranteed to be lowercase)